Add win32 for Windows clipboard

thanhph111 opened the pull request.

b-init committed:
Re-did windows clipboard using clipette

This is much faster. Basically instant.

Ditched support for bitmap in `push` because blender doesn't completely support bitmaps and I think we don't need it anyways.

thanhph111 committed:
Reformat a little bit

thanhph111 committed:
Merge branch 'main' into win32-windows-clipboard

Yeetus3141 commented:
Here are some documentation and resources I used while making clipette:

[win32 cliboard functions](https://learn.microsoft.com/en-us/windows/win32/dataxchg/clipboard)
[win32 memory management functions](https://learn.microsoft.com/en-us/windows/win32/memory/memory-management-functions)
[`CF_HDROP` format](https://learn.microsoft.com/en-us/windows/win32/shell/clipboard#cf_hdrop)
[ctypes wintypes reference](https://epydoc.sourceforge.net/stdlib/ctypes.wintypes-module.html)
[ctypes docs](https://docs.python.org/3/library/ctypes.html)
[bitmap image file structure](https://en.wikipedia.org/wiki/BMP_file_format#File_structure)


This is a very crude way of using the clipboard and python doesn't know what's going on when dealing with these win32 modules so error management might end up being a challenge. If any of the win32 functions spit any error, we must use [GetLastError](https://learn.microsoft.com/en-us/windows/win32/api/errhandlingapi/nf-errhandlingapi-getlasterror) to access it but I haven't implemented that so far.

b-init committed:
updated clipette.py to prevent improper exit

to prevent exiting without closing clipboard

b-init committed:
don't use any color transformation while converting images

• `image.save()` doesn't consider the updated format (bug?) so we're stuck with `save_render()` which uses render settings and would transform image accordingly (often set to 'Filmic' colorspace) which is unwanted during conversion. Fixed that by settings render settings to standard values temporarily. Fixes #47
• render is saved in `get_save_directory()` instead of `tempdir`
